client/data/media/use-delete-media-mutation.js

```diff
@@ -0,0 +1,19 @@
+import { useCallback } from 'react';
+import { useMutation } from 'react-query';
+import wp from 'calypso/lib/wp';
+
+export const useDeleteMediaMutation = ( queryOptions = {} ) => {
+	const mutation = useMutation(
+		( { siteId, mediaId } ) => wp.req.post( `/sites/${ siteId }/media/${ mediaId }/delete` ),
+		{ ...queryOptions }
```

Comment:
syntax nit: do we need to shallow-clone the `queryOptions` here? Or can we simply pass `queryOption` as it is, without defaulting to `{}` and cloning?

Reply:
Nope, we could just pass it on. Good call 👍🏻